### ISSUE TITLE
Beginner tutorial

### ISSUE THREAD

**Issue by GuidoBartoli** [keras-rl/keras-rl#258]
Hi to everybody,
Some weeks ago I posted this question on keras-rl google groups forum, but nobody answered... I try to repost it here hoping to receive some feedback :)

I worked some months on a deep learning project applied to computer vision and I gained some knowledge on CNNs and the traditional preprocessing/training/validation/test steps of machine learning workflow.
Now I would like to apply deep learning to another personal project: train a RNN using reinforcement learning to build an agent for a simple card game.
Since I worked with Keras+Tensorflow framework, I thought keras-rl package would be a good choice for experiments.
I installed it with all dependencies on my computer and I successfully ran some example scripts with gym, everything works, visualization included. 
I took a look at the example scripts, but all of them refer to the gym environment... I would like to know where to start to build a new environment from scratch, since the game I'm working is custom designed and I have to manually setup actions, observation, reward, etc...

Can anyone point me some tutorial or documentation on how to getting started with keras-rl for a custom environment?

Many thanks,
Best regards

**Comment by neilslater** [comment]
Probably the simplest thing to do is write an API to your environment that *presents*  it in a similar way to a gym environment. That would allow you to use the example scripts with little modification.

I.e. it should `step`, `reset` and `render` methods, plus set `action_space`, `observation_space`, `reward_range` and `state_size` attributes. It is possible to use gym base classes and utilities to do this. See https://stackoverflow.com/questions/45068568/is-it-possible-to-create-a-new-gym-environment-in-openai for some discussion and examples.

**Comment by mirraaj** [comment]
Hey @GuidoBartoli, as @neilslater pointed out you need to develop a gym-like interface for the agent to interact with. You may wish to have a look at https://github.com/openai/gym/tree/master/gym/envs#how-to-create-new-environments-for-gym.

Render could be skipped if visualization is not necessary to demonstrate the result.

**Comment by GuidoBartoli** [comment]
Ok, many thanks for the information, I will take a look at the gym interface and try to provide the same outputs!

Every turn the agent has at most 10 cards to choose from (action space?) after having observed the other players plays (observation space?), while the reward is gaining more points than the opponents (the points are merged with the player sitting on the opposite table side, two teams are playing).

**Comment by GuidoBartoli** [comment]
Actually I thought to train only one "smart" agent to play against "simple" ones that use only greedy or statistical methods. Theoretically, leaving the smart one play many times against the others and learning from its mistakes to maximize the points, it should gradually evolve from random play to easily beating the others. Do you have any suggestions if this methodology is feasible for this kind of application?

**Comment by mirraaj** [comment]
@GuidoBartoli , I think by using the term "greedy" and "statistical" you define your opposite agents to follow some standard policy. Under such scenario, the above tutorial on how to make an environment using gym should be enough for you. I do not have any suggestion in particular. However, I will suggest you define the system properly. Remember the fact that Reinforcement Learning algorithms work best with MDPs. Hence, your definition of environment plays a vital role in what policy your agent learns.

**Comment by GuidoBartoli** [comment]
Ok, thanks to everybody, I will let you know how it goes when I will have a working first version :)
